include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2120,116 +1721,394 @@ namespace internal
-            (EvaluationFlags::gradients | EvaluationFlags::hessians))
-          FEEvaluationImplCollocation<dim, n_q_points_1d - 1, Number>::
-            do_integrate(shape_data,
-                         integration_flag & (EvaluationFlags::gradients |
-                                             EvaluationFlags::hessians),
-                         fe_eval.begin_values() + c * n_q_points,
-                         fe_eval.begin_gradients() + c * dim * n_q_points,
-                         fe_eval.begin_hessians() +
-                           c * dim * (dim + 1) / 2 * n_q_points,
-                         /*add_into_values_array=*/
-                         integration_flag & EvaluationFlags::values);
-
-        // transform back to the original space
-        FEEvaluationImplBasisChange<
-          evaluate_evenodd,
-          EvaluatorQuantity::value,
-          dim,
-          (fe_degree >= n_q_points_1d ? n_q_points_1d : fe_degree + 1),
-          n_q_points_1d>::do_backward(1,
-                                      shape_data.shape_values_eo,
-                                      add_into_values_array,
-                                      fe_eval.begin_values() + c * n_q_points,
-                                      values_dofs +
-                                        c * Utilities::pow(fe_degree + 1, dim));
+        const bool do_values = evaluation_flag & EvaluationFlags::values;
+        if ((evaluation_flag & EvaluationFlags::gradients) != 0u)
+          integrate_gradients_collocation<n_q_points_1d, dim>(shape_data[0],
+                                                              values,
+                                                              gradients,
+                                                              do_values);
+        if constexpr (dim > 2)
+          work_tangential<2, 0, false>(shape_data[1], values, values);
+        work_tangential<1, 0, false>(shape_data[1], values, values);
+        work_normal<0, false>(shape_data[0],
+                              values,
+                              values_dofs,
+                              add_into_values_array);
+
+        values += n_points;
+        gradients += n_points * dim;
+        values_dofs += dofs_per_component;
+
+        if ((evaluation_flag & EvaluationFlags::gradients) != 0u)
+          integrate_gradients_collocation<n_q_points_1d, dim>(shape_data[0],
+                                                              values,
+                                                              gradients,
+                                                              do_values);
+        if constexpr (dim > 2)
+          work_tangential<2, 1, false>(shape_data[1], values, values);
+        work_tangential<0, 1, false>(shape_data[1], values, values);
+        work_normal<1, false>(shape_data[0],
+                              values,
+                              values_dofs,
+                              add_into_values_array);
```

Comment:
This is the main code to perform the Raviart-Thomas evaluation now (with additional code below for 3d and the other two components, which need to be rolled out manually). As I use the collocation derivative technique (aka basis change to a diagonal interpolation matrix), I only need to perform anisotropic interpolations for the values. I think this is more concise than the previous evaluators, and it is also more easily extensible to H(curl) evaluators (where we switch the high degree and low degree, that can be done by the same class but with different parameters).